internal/rpc/rpc_handler.go

```diff
@@ -0,0 +1,172 @@
+package rpc
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+
+	hadoop "github.com/colinmarc/hdfs/v2/internal/protocol/hadoop_common"
+	"github.com/golang/protobuf/proto"
+	"gopkg.in/jcmturner/gokrb5.v5/crypto"
+	"gopkg.in/jcmturner/gokrb5.v5/gssapi"
+	"gopkg.in/jcmturner/gokrb5.v5/iana/keyusage"
+	krbtypes "gopkg.in/jcmturner/gokrb5.v5/types"
+)
+
+// To check interface validity
+var rpcWriter RpcWriter = &BasicRpcWriter{}
+var rpcReader RpcReader = &BasicRpcReader{}
+var saslRpcReader RpcReader = &SaslRpcReader{}
+
+// RpcWriter is an interface for sending RPC payload
+type RpcWriter interface {
+	WriteRequest(w io.Writer, method string, requestID int32, req proto.Message) error
+}
+
+// RpcReader is an interface for receiving RPC payload
+type RpcReader interface {
+	ReadResponse(r io.Reader, method string, requestID int32, resp proto.Message) error
+}
+
+// BasicRpcWriter is a basic RPC writer
+type BasicRpcWriter struct {
+	// ClientID is the client ID of this writer
+	ClientID []byte
+}
+
+// BasicRpcReader is a basic RPC reader
+type BasicRpcReader struct {
+}
+
+// SaslRpcReader is a RPC reader which wrap payload with SASL
```

Comment:
Sorry for being picky, but can you change all these comments to be complete sentences?

Reply:
I will try it.